src/Compilers/Core/Portable/Compilation/CompilationContext.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+
+namespace Microsoft.CodeAnalysis.Collections
+{
+    /// <summary>
+    /// A mutable typesafe dictionary from a pair of compilation and key (of type <see cref="CompilationContext.Key{T}"/>)
+    /// to a value (of type T). All keys and values associated with a compilation are reachable for GC purposes as a
+    /// consequence of this API for only as long as the compilation is reachable.
+    /// </summary>
+    public class CompilationContext
+    {
+        private static CompilationContext _instance = new CompilationContext();
+        public static CompilationContext Instance => _instance;
```

Comment:
Is there a reason to have an instance of this type at all? Should the type be static?